torch_ttnn/passes/memory_pass.py

```diff
@@ -0,0 +1,240 @@
+import torch
+import ttnn
+from torch.fx.passes.infra.pass_base import PassBase, PassResult
+from torch_ttnn.passes.lowering.add_data_move_pass import is_tt_compute
+
+from torch_ttnn.mem_utils import *
+
+
+def memory_footprint(gm: torch.fx.GraphModule) -> torch.fx.GraphModule:
```

Comment:
This function is way too big and really hard to navigate and understand from a quick glance.
Can you please break it down into smaller several functions each representing some logical block?